src/backends/kms.rs

```
@@ -224,6 +224,29 @@ impl<D: HasDisplayHandle + ?Sized, W: HasWindowHandle> SurfaceInterface<D, W> fo
         let front_buffer = SharedBuffer::new(&self.display, width, height)?;
         let back_buffer = SharedBuffer::new(&self.display, width, height)?;
 
+        // Iterate over all modes of all the connectors,
+        // and pick one that is compatible with the size given by the user.
+        let mode = self
+            .connectors
+            .iter()
+            .flat_map(|con| self.display.get_modes(*con).unwrap())
```

Any chance we can bubble this error up instead of `unwrap`ping on it?